File: server/backend/internal/models/achievement.go
package models

type AchievementModel struct {
	ID      int64  `gorm:"primaryKey" json:"id"`
	Type    string `json:"type"`
	TeamID  int64  `json:"teamId"`
	EventID int64  `json:"eventId"`
	TrackID int64  `json:"trackId"`
	Result  string `json:"result"`
}

File: client/src/entities/Achievement/model/types/Achievement.ts
import { EventType } from '@/entities/Event';

export type ResultType = 'first' | 'second' | 'third' | 'participant';

export interface Achievement {
    id: number;

    teamId: number;
    userId: number;

    eventName: string;
    eventType: EventType;
    teamTitle: string;
    eventId: number;
    trackId?: number;

    result: ResultType;
}
